src/DynamoDb/SetValue.php

```diff
@@ -0,0 +1,116 @@
+<?php
+namespace Aws\DynamoDb;
+
+use GuzzleHttp\ToArrayInterface;
+
+/**
+ * Special object to represent a DynamoDB set (SS/NS/BS) value.
+ */
+class SetValue implements \JsonSerializable, \ArrayAccess, \Countable, \IteratorAggregate, ToArrayInterface
+{
+    /** @var array Values in the set. */
+    private $values;
+
+    /** @var string Set type. One of "SS", "NS", or "BS. */
+    private $type;
+
+    /**
+     * @param string $type   One of "SS", "NS", or "BS.
+     * @param array  $values Values in the set.
+     */
+    public function __construct($type, array $values = [])
+    {
+        $this->type = $type;
+        if (!in_array($this->type, ['SS', 'NS', 'BS'], true)) {
+            throw new \InvalidArgumentException(
+                'Invalid set type. Must be BS, NS, or SS'
+            );
+        }
+
+        foreach ($values as $value) {
+            $this->offsetSet(null, $value);
+        }
+    }
+
+    /**
+     * Get the set type.
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * Get the values formatted for DynamoDB.
+     *
+     * @return array
+     */
+    public function getValues()
+    {
+        if (empty($this->values)) {
+            throw new \RuntimeException('DynamoDB does not allow empty sets.');
+        }
+
+        return array_map('strval', array_keys($this->values));
```

Comment:
You would think that, but PHP treats numerical keys as numbers no matter how hard you try to make them strings.